substrate/frame/multi-asset-bounties/src/lib.rs

```diff
@@ -0,0 +1,1962 @@
+			Some(curator.clone())
+		} else {
+			None
+		};
+		match child_bounty_id {
+			None => Ok((
+				parent_bounty.asset_kind,
+				parent_bounty.value,
+				parent_bounty.fee,
+				parent_bounty.curator_deposit,
+				parent_bounty.status,
+				parent_curator,
+			)),
+			Some(child_bounty_id) => {
+				let child_bounty = ChildBounties::<T, I>::get(parent_bounty_id, child_bounty_id)
+					.ok_or(Error::<T, I>::InvalidIndex)?;
+				Ok((
+					parent_bounty.asset_kind,
+					child_bounty.value,
+					child_bounty.fee,
+					child_bounty.curator_deposit,
+					child_bounty.status,
+					parent_curator,
+				))
+			},
+		}
+	}
+
+	/// Updates the status and optionally the fee and curator deposit of a child-/bounty.
+	pub fn update_bounty_details(
+		parent_bounty_id: BountyIndex,
+		child_bounty_id: Option<BountyIndex>,
+		new_status: BountyStatus<T::AccountId, PaymentIdOf<T, I>, T::Beneficiary>,
+		maybe_fee: Option<BalanceOf<T, I>>,
+		maybe_curator_deposit: Option<BalanceOf<T, I>>,
+	) -> Result<(), DispatchError> {
+		match child_bounty_id {
+			None => {
+				let mut bounty =
+					Bounties::<T, I>::get(parent_bounty_id).ok_or(Error::<T, I>::InvalidIndex)?;
+				bounty.status = new_status;
+				if let Some(curator_deposit) = maybe_curator_deposit {
+					bounty.curator_deposit = curator_deposit;
+				}
+				if let Some(fee) = maybe_fee {
+					bounty.fee = fee;
+				}
+				Bounties::<T, I>::insert(parent_bounty_id, bounty);
+				Ok(())
+			},
+			Some(child_bounty_id) => {
+				let mut bounty = ChildBounties::<T, I>::get(parent_bounty_id, child_bounty_id)
+					.ok_or(Error::<T, I>::InvalidIndex)?;
+				bounty.status = new_status;
+				if let Some(curator_deposit) = maybe_curator_deposit {
+					bounty.curator_deposit = curator_deposit;
+				}
+				if let Some(fee) = maybe_fee {
+					bounty.fee = fee;
+				}
+				ChildBounties::<T, I>::insert(parent_bounty_id, child_bounty_id, bounty);
+				Ok(())
+			},
+		}
+	}
+
+	/// Calculates amount the beneficiary and curator stash during child-/bounty payout.
+	fn calculate_curator_fee_and_payout(
+		parent_bounty_id: BountyIndex,
+		fee: BalanceOf<T, I>,
+		value: BalanceOf<T, I>,
+	) -> (BalanceOf<T, I>, BalanceOf<T, I>) {
+		// Get total child bounties curator fees, and subtract it from the parent
+		// curator fee (the fee in present referenced bounty, `self`).
+		let children_fee = ChildBountiesCuratorFeesPerParent::<T, I>::get(parent_bounty_id);
+		debug_assert!(children_fee <= fee);
+		let final_fee = fee.saturating_sub(children_fee);
+
+		// Get total child bounties value, and subtract it from the parent
+		// value (the value in present referenced bounty, `self`).
+		let children_value = ChildBountiesValuePerParent::<T, I>::get(parent_bounty_id);
+		debug_assert!(children_value <= value);
+		let value_remaining = value.saturating_sub(children_value);
+		let payout = value_remaining.saturating_sub(final_fee);
+
+		(final_fee, payout)
+	}
```

Comment:
I don't understand this function, we call it in `do_process_payout_payments` which is called in `award_bounty` to award child or parent bounty.

But here we only compute the fee/value for parent bounty no?
if it is for a child bounty, then `fee` will be less than `children_fee`, and we shouldn't do the substraction. same for value.